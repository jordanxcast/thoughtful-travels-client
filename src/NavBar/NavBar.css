.NavBar {
  display: flex;
  flex-direction: row;
  position: fixed;
  top: 0;
  width: 100%;
  border: 1px solid black;
  padding: 20px;
  justify-content: space-between;
  background-color: #131B23;
  color: #F2F7F2;
  box-shadow: 0px 6px 5px 0px rgba(0,0,0,0.75);
  z-index: 12;
}

.NavBar-link {
  text-decoration: none;
  color: #F2F7F2;
}

.NavBar-link:active {
  color: #656176;
}

.NavBar-link:hover {
  color: #656176;
}

.NavBar_not-logged-in > .NavBar-link {
  margin-left: 20px;
}