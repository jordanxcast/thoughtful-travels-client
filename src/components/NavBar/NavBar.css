.NavBar {
  display: flex;
  flex-direction: row;
  position: fixed;
  top: 0;
  width: 100%;
  border: 1px solid #7098a9;
  padding: 20px;
  justify-content: space-between;
  background-color: #7098a9;
  color: #f2f7f2;
  /* box-shadow: 0px 6px 5px 0px rgba(0, 0, 0, 0.75); */
  z-index: 12;
}

.NavBar-link {
  text-decoration: none;
  color: #f2f7f2;
}

.NavBar-link:active {
  color: #656176;
  border-bottom: 2px solid #656176;
}

.NavBar-link:hover {
  color: #656176;
  border-bottom: 2px solid #656176;
}

.NavBar_not-logged-in > .NavBar-link {
  margin-left: 20px;
}
