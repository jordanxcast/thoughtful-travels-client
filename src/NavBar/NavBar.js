import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../services/token-service'
import ApiContext from '../ApiContext'
import './NavBar.css'

export default class NavBar extends Component {
  constructor(props) {
    super(props)
    this.state = {
      loggedIn: TokenService.hasAuthToken()
    }
  }
  static contextType = ApiContext

  componentDidMount() {
    console.log('loggedIn:', this.state.loggedIn)
    console.log('has auth token:', TokenService.hasAuthToken())
    if(this.context.authToken || TokenService.hasAuthToken()) {
      this.setState({
        loggedIn: true
      })
    } else {
      this.setState({
        loggedIn: false
      })
    }
  }

 
  
  handleLogoutClick = () => {
    TokenService.clearAuthToken()
    this.context.handleAuthToken(null)
    this.setState({
      loggedIn: false
    })
  }
  
  renderLogoutLink() {
    return (
      <div className='NavBar__logged-in'>
        <Link
          className='NavBar-link'
          onClick={this.handleLogoutClick}
          to='/'>
          Logout
        </Link>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='NavBar_not-logged-in'>
        <Link
          className='NavBar-link'
          to='/login'>
          Login
        </Link>
      </div>
    )
  }

  render() {
    
    return (
      <>
        <nav className='NavBar'>
          <div className='NavBar-logo'>
            <Link to='/' className='NavBar-link'>
              {' '}
              <i className="fa far fa-compass"></i>

              TT
            </Link>
          </div>
          <div className='NavBar-menu'>
            {
              this.context.authToken 
                ? this.renderLogoutLink()
                : this.renderLoginLink()
            }
          </div>
        </nav>
      </>
    )
  }
}
