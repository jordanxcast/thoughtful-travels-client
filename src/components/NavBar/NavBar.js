import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faCompass} from '@fortawesome/free-solid-svg-icons'
import TokenService from '../../services/token-service'
import ApiContext from '../../ApiContext'
// import IdleService from '../../services/idle-service'
import './NavBar.css'

export default class NavBar extends Component {
  constructor(props) {
    super(props)
    this.state = {
      loggedIn: TokenService.hasAuthToken()
    }
  }
  static contextType = ApiContext

  componentDidMount() {
    if(this.context.authToken || TokenService.hasAuthToken()) {
      this.setState({
        loggedIn: true
      })
    } else {
      this.setState({
        loggedIn: false
      })
    }
  }

  handleLogoutClick = () => {
    TokenService.clearAuthToken()
    this.context.handleAuthToken(null)
    this.setState({
      loggedIn: false
    })
    // TokenService.clearCallbackBeforeExpiry()
    // IdleService.unRegisterIdleResets()
  }
  
  renderLogoutLink() {
    return (
      <div className='NavBar__logged-in'>
        <Link
          className='NavBar-link'
          onClick={this.handleLogoutClick}
          to='/'>
          Logout
        </Link>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='NavBar_not-logged-in'>
        <Link
          className='NavBar-link'
          to='/login'>
          Login
        </Link>
      </div>
    )
  }

  render() {
    
    return (
      <>
        <nav className='NavBar'>
          <div className='NavBar-logo'>
            <Link to='/' className='NavBar-link'>
              <FontAwesomeIcon icon={faCompass} className='NavBar-icon'/>
              {' '}
              TT
            </Link>
          </div>
          <div className='NavBar-menu'>
            {
              this.context.authToken 
                ? this.renderLogoutLink()
                : this.renderLoginLink()
            }
          </div>
        </nav>
      </>
    )
  }
}
