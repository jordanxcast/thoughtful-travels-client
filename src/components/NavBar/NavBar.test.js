import React from 'react'
import NavBar from './NavBar'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'


describe(`<NavBar />`, () => {
  //Snapshot Testing w/ Enzyme
  it('renders UI as expected', () => {
    const wrapper = shallow(<NavBar />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})