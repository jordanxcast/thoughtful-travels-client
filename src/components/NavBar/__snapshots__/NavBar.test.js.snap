// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders UI as expected 1`] = `
<Fragment>
  <nav
    className="NavBar"
  >
    <div
      className="NavBar-logo"
    >
      <Link
        className="NavBar-link"
        to="/"
      >
        <FontAwesomeIcon
          border={false}
          className="NavBar-icon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                496,
                512,
                Array [],
                "f14e",
                "M225.38 233.37c-12.5 12.5-12.5 32.76 0 45.25 12.49 12.5 32.76 12.5 45.25 0 12.5-12.5 12.5-32.76 0-45.25-12.5-12.49-32.76-12.49-45.25 0zM248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm126.14 148.05L308.17 300.4a31.938 31.938 0 0 1-15.77 15.77l-144.34 65.97c-16.65 7.61-33.81-9.55-26.2-26.2l65.98-144.35a31.938 31.938 0 0 1 15.77-15.77l144.34-65.97c16.65-7.6 33.8 9.55 26.19 26.2z",
              ],
              "iconName": "compass",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         
        TT
      </Link>
    </div>
    <div
      className="NavBar-menu"
    >
      <div
        className="NavBar_not-logged-in"
      >
        <Link
          className="NavBar-link"
          to="/login"
        >
          Login
        </Link>
      </div>
    </div>
  </nav>
</Fragment>
`;
